driver/src/orderbook/streamed/balance.rs

```diff
@@ -60,137 +32,124 @@ impl Balance {
+        self.sell_buy_internal(bigint_u256::u256_to_bigint(fee), batch_id)
     }
 
     pub fn revert_solution_submission(&mut self, fee: U256, batch_id: BatchId) -> Result<()> {
-        self.revert_sell_buy_internal(fee, batch_id, TradeType::Buy)
+        self.revert_sell_buy_internal(bigint_u256::u256_to_bigint(fee), batch_id)
     }
 
-    fn sell_buy_internal(
-        &mut self,
-        amount: U256,
-        batch_id: BatchId,
-        operation: TradeType,
-    ) -> Result<()> {
+    /// Adds the amount to the proceeds.
+    fn sell_buy_internal(&mut self, amount: BigInt, batch_id: BatchId) -> Result<()> {
         ensure!(self.proceeds.batch_id <= batch_id, "trade for past batch");
-        self.apply_existing_deposit_and_proceeds(batch_id)?;
+        self.apply_existing_deposit_and_proceeds(batch_id);
         // Now old proceeds have been cleared. If there is an existing proceed for this batch then
         // setting the batch_id does nothing and we add to the field.
         self.proceeds.batch_id = batch_id;
-        let field = self.proceeds.get_field(operation);
-        *field = field
-            .checked_add(amount)
-            .ok_or_else(|| anyhow!("math overflow"))?;
+        self.proceeds.amount += amount;
         Ok(())
     }
 
-    fn revert_sell_buy_internal(
-        &mut self,
-        amount: U256,
-        batch_id: BatchId,
-        operation: TradeType,
-    ) -> Result<()> {
+    /// Subtracts the amount from the proceeds.
+    fn revert_sell_buy_internal(&mut self, amount: BigInt, batch_id: BatchId) -> Result<()> {
         ensure!(
             self.proceeds.batch_id == batch_id,
             "reverting non existent trade"
         );
-        let field = self.proceeds.get_field(operation);
-        *field = field
-            .checked_sub(amount)
-            .ok_or_else(|| anyhow!("math underflow"))?;
+        self.proceeds.amount -= amount;
         Ok(())
     }
 
-    fn balance_with_deposit_and_proceeds(&self, current_batch_id: BatchId) -> Result<U256> {
-        let mut balance = self.balance;
-        if self.deposit.batch_id < current_batch_id {
-            balance = balance
-                .checked_add(self.deposit.amount)
-                .ok_or_else(|| anyhow!("math overflow"))?;
+    fn balance_with_deposit_and_proceeds(&self, current_batch_id: BatchId) -> BigInt {
+        let mut result = self.balance.clone();
+        if let Some(deposit) = self.deposit.amount(current_batch_id) {
+            result += deposit
         }
-        if self.proceeds.batch_id < current_batch_id {
-            balance = balance
-                .checked_add(self.proceeds.increase)
-                .ok_or_else(|| anyhow!("math overflow"))?
-                .checked_sub(self.proceeds.decrease)
-                .ok_or_else(|| anyhow!("math underflow"))?;
+        if let Some(proceed) = self.proceeds.amount(current_batch_id) {
+            result += proceed
```

Comment:
I can see the benefit of this cleanup... Actually almost seems like it would have been easier to do it this way in the first place.